source/base/ThreadPool.ooc

```diff
@@ -70,7 +76,15 @@ _TaskPromise: class extends Promise {
 		this _task free()
 		super()
 	}
-	wait: override func -> Bool { this _task wait() }
+	wait: func -> Bool { this _task wait() }
+	wait: func ~timeout (seconds: Double) -> Bool {
+		timer := ClockTimer new() . start()
+		status := false
+		while (timer stop() / 1000.0 < seconds && !status)
```

Comment:
No, this works. But yes, I agree with you, it's ugly. I'll add some `sleep` at least.